Error compiling on Raspbery pi 3 model b+

Hello!
First of all, i apologize for my bad English.
I had a problem and i can't understand, how to solve it.
I attach the log.

**cd build && cmake ..**

```
-- The CXX compiler identification is GNU 6.3.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Boost version: 1.62.0
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29") 
-- Checking for module 'capstone>=3.0.4'
--   Found capstone, version 3.0.4
-- Checking for module 'libgvc>=2.38.0'
--   Found libgvc, version 2.38.0
-- Checking for module 'double-conversion'
-- Looking for C++ include double-conversion/double-conversion.h
-- Looking for C++ include double-conversion/double-conversion.h - not found
CMake Warning at CMakeLists.txt:66 (message):
  libdouble-conversion header wasn't found.  32- and 64-bit floating-point
  values will be shown with max_digits10 digits of precision instead of
  shortest representation.
-- Configuring done
-- Generating done
```

**make** 

```
...
Generating qrc source edb_automoc.dir/qrc_debugger_FCYOZ6RSSZA4NC.cpp
[  1%] Built target edb_automoc
Scanning dependencies of target edb
[  2%] Building CXX object src/CMakeFiles/edb.dir/BasicBlock.cpp.o
In file included from /home/pi/Desktop/edb-debugger/include/Instruction.h:19:0,
                 from /home/pi/Desktop/edb-debugger/include/arch/arm-generic/ArchTypes.h:22,
                 from /home/pi/Desktop/edb-debugger/include/Types.h:44,
                 from /home/pi/Desktop/edb-debugger/include/BasicBlock.h:23,
                 from /home/pi/Desktop/edb-debugger/src/BasicBlock.cpp:19:
/home/pi/Desktop/edb-debugger/include/../src/capstone-edb/include/Instruction.h:158:3: error: #error "Not implemented"
 # error "Not implemented"
   ^~~~~
src/CMakeFiles/edb.dir/build.make:62: recipe for target 'src/CMakeFiles/edb.dir/BasicBlock.cpp.o' failed
```

```
Linux raspberrypi 4.14.98-v7+ #1200 SMP Tue Feb 12 20:27:48 GMT 2019 armv7l GNU/Linux

/proc/cpuinfo

processor	: 3
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 89.60
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

Hardware	: BCM2835
Revision	: a020d3
Serial		: 00000000e5caf913
```

No :(
Any ideas?

[ 18%] Building CXX object src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o
/home/pi/Desktop/edb-debugger/src/RegisterViewModelBase.cpp: In member function ‘int RegisterViewModelBase::SIMDFormatItem<StoredType, SizingType>::valueMaxLength() const [with StoredType = edb::detail::value_type80; SizingType = edb::detail::value_type80]’:
/home/pi/Desktop/edb-debugger/src/RegisterViewModelBase.cpp:934:2: error: static assertion failed: Only 80-bit long doubles are supported
  static_assert(sizeof(edb::value80) <= sizeof(long double), "Only 80-bit long doubles are supported");
  ^~~~~~~~~~~~~
src/CMakeFiles/edb.dir/build.make:806: recipe for target 'src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o' failed
make[2]:  [src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o] Error 1
CMakeFiles/Makefile2:88: recipe for target 'src/CMakeFiles/edb.dir/all' failed
make[1]:  [src/CMakeFiles/edb.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make:  [all] Error 2

git clone --recursive https://github.com/eteran/edb-debugger.git
git checkout da26f58f09cf087d103890dec511fefe8fa17e90
cmake ..

[  3%] Building CXX object src/CMakeFiles/edb.dir/CommentServer.cpp.o
In file included from /home/pi/Desktop/edb-debuggerCHECKOUT/src/CommentServer.cpp:19:0:
/home/pi/Desktop/edb-debuggerCHECKOUT/src/CommentServer.h:29:79: error: expected class-name before ‘{’ token
 class CommentServer : public QObject, public QHexView::CommentServerInterface {
                                                                               ^
/home/pi/Desktop/edb-debuggerCHECKOUT/src/CommentServer.h:37:10: error: ‘void CommentServer::set_comment(QHexView::address_t, const QString&)’ marked ‘override’, but does not override
     void set_comment(QHexView::address_t address, const QString &comment) override;
          ^~~~~~~~~~~
/home/pi/Desktop/edb-debuggerCHECKOUT/src/CommentServer.h:38:13: error: ‘QString CommentServer::comment(QHexView::address_t, int) const’ marked ‘override’, but does not override
     QString comment(QHexView::address_t address, int size) const override;
             ^~~~~~~
/home/pi/Desktop/edb-debuggerCHECKOUT/src/CommentServer.h:39:10: error: ‘void CommentServer::clear()’ marked ‘override’, but does not override
     void clear() override;
          ^~~~~
src/CMakeFiles/edb.dir/build.make:134: recipe for target 'src/CMakeFiles/edb.dir/CommentServer.cpp.o' failed
make[2]:  [src/CMakeFiles/edb.dir/CommentServer.cpp.o] Error 1
CMakeFiles/Makefile2:86: recipe for target 'src/CMakeFiles/edb.dir/all' failed
make[1]:  [src/CMakeFiles/edb.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make:  [all] Error 2



git clone --recursive https://github.com/eteran/edb-debugger.git
rm CMakeLists.txt && wget https://raw.githubusercontent.com/eteran/edb-debugger/da26f58f09cf087d103890dec511fefe8fa17e90/CMakeLists.txt
cmake ..
[ 18%] Building CXX object src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o
/home/pi/Desktop/edb-debuggerMASTER/src/RegisterViewModelBase.cpp: In member function ‘int RegisterViewModelBase::SIMDFormatItem<StoredType, SizingType>::valueMaxLength() const [with StoredType = edb::detail::value_type80; SizingType = edb::detail::value_type80]’:
/home/pi/Desktop/edb-debuggerMASTER/src/RegisterViewModelBase.cpp:934:2: error: static assertion failed: Only 80-bit long doubles are supported
  static_assert(sizeof(edb::value80) <= sizeof(long double), "Only 80-bit long doubles are supported");
  ^~~~~~~~~~~~~

src/CMakeFiles/edb.dir/build.make:806: recipe for target 'src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o' failed
make[2]:  [src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o] Error 1
CMakeFiles/Makefile2:88: recipe for target 'src/CMakeFiles/edb.dir/all' failed
make[1]:  [src/CMakeFiles/edb.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make:  [all] Error 2

Thx! Any ideas?

git clone --recursive https://github.com/10110111/edb-debugger

[ 22%] Building CXX object src/CMakeFiles/edb.dir/capstone-edb/Instruction.cpp.o
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::is_simd_register(const CapstoneEDB::Operand&)’:
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:45:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type != X86_OP_REG)
                       ^~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::apriori_not_simd(const CapstoneEDB::Instruction&, const CapstoneEDB::Operand&)’:
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:65:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_REG && !is_simd_register(operand))
                       ^~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:67:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_IMM)
                       ^~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::KxRegisterPresent(const CapstoneEDB::Instruction&)’:
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:78:19: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (op->type == X86_OP_REG && X86_REG_K0 <= op->reg && op->reg <= X86_REG_K7) {
                   ^~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp: In member function ‘std::__cxx11::string CapstoneEDB::Formatter::to_string(const CapstoneEDB::Operand&) const’:
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:441:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_REG) {
                       ^~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::KxRegisterPresent(const CapstoneEDB::Instruction&)’:
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:479:19: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (op->type == X86_OP_REG && X86_REG_K0 <= op->reg && op->reg <= X86_REG_K7)
                   ^~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::is_SIMD_PD(const CapstoneEDB::Operand&)’:
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:829:24: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (insn[2]->type != X86_OP_IMM)
                        ^~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::modifies_pc(const CapstoneEDB::Instruction&)’:
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:1100:9: error: ‘const struct cs_arm_op’ has no member named ‘access’
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
         ^~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:1100:17: error: ‘CS_AC_WRITE’ was not declared in this scope
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
                 ^~~~~~~~~~~
/home/pi/Desktop/newfork/edb-debugger/src/capstone-edb/Instruction.cpp:1100:41: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum cs_op_type’ [-Wenum-compare]
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)

```
[ 16%] Building CXX object src/CMakeFiles/edb.dir/Register.cpp.o
cd /home/pi/edb-debugger/build/src && /usr/bin/c++   -DDEFAULT_PLUGIN_PATH=\"/usr/local/lib/edb\" -DEDB_ARM32 -DEDB_IS_32_BIT=true -DEDB_IS_64_BIT=false -DENABLE_GRAPH -DGIT_BRANCH=a3dcab4d849f9bb2d9dcde7a6acf1af4ca87d2dd -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XMLPATTERNS_LIB -DQT_XML_LIB -Dedb_EXPORTS -I/home/pi/edb-debugger/build/src -I/home/pi/edb-debugger/src -I/usr/include/capstone -I/usr/include/graphviz -I/home/pi/edb-debugger/include -I/home/pi/edb-debugger/include/os/unix -I/home/pi/edb-debugger/include/os/unix/linux -I/home/pi/edb-debugger/include/arch/arm-generic -I/home/pi/edb-debugger/src/capstone-edb -I/home/pi/edb-debugger/src/qhexview -I/home/pi/edb-debugger/src/widgets -I/home/pi/edb-debugger/src/session -isystem /usr/include/arm-linux-gnueabihf/qt5 -isystem /usr/include/arm-linux-gnueabihf/qt5/QtWidgets -isystem /usr/include/arm-linux-gnueabihf/qt5/QtGui -isystem /usr/include/arm-linux-gnueabihf/qt5/QtCore -isystem /usr/lib/arm-linux-gnueabihf/qt5/mkspecs/linux-g++ -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXml -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXmlPatterns -isystem /usr/include/arm-linux-gnueabihf/qt5/QtNetwork -isystem /usr/include/arm-linux-gnueabihf/qt5/QtSvg  -O2 -g -DNDEBUG   -W -Wall -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -pedantic -Wdouble-promotion -Wformat=2 -Wno-unused-macros -Wno-switch-enum -Wno-unknown-pragmas -Wlogical-op -Wsuggest-override -fPIC -std=c++14 -o CMakeFiles/edb.dir/Register.cpp.o -c /home/pi/edb-debugger/src/Register.cpp
[ 16%] Building CXX object src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o
cd /home/pi/edb-debugger/build/src && /usr/bin/c++   -DDEFAULT_PLUGIN_PATH=\"/usr/local/lib/edb\" -DEDB_ARM32 -DEDB_IS_32_BIT=true -DEDB_IS_64_BIT=false -DENABLE_GRAPH -DGIT_BRANCH=a3dcab4d849f9bb2d9dcde7a6acf1af4ca87d2dd -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XMLPATTERNS_LIB -DQT_XML_LIB -Dedb_EXPORTS -I/home/pi/edb-debugger/build/src -I/home/pi/edb-debugger/src -I/usr/include/capstone -I/usr/include/graphviz -I/home/pi/edb-debugger/include -I/home/pi/edb-debugger/include/os/unix -I/home/pi/edb-debugger/include/os/unix/linux -I/home/pi/edb-debugger/include/arch/arm-generic -I/home/pi/edb-debugger/src/capstone-edb -I/home/pi/edb-debugger/src/qhexview -I/home/pi/edb-debugger/src/widgets -I/home/pi/edb-debugger/src/session -isystem /usr/include/arm-linux-gnueabihf/qt5 -isystem /usr/include/arm-linux-gnueabihf/qt5/QtWidgets -isystem /usr/include/arm-linux-gnueabihf/qt5/QtGui -isystem /usr/include/arm-linux-gnueabihf/qt5/QtCore -isystem /usr/lib/arm-linux-gnueabihf/qt5/mkspecs/linux-g++ -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXml -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXmlPatterns -isystem /usr/include/arm-linux-gnueabihf/qt5/QtNetwork -isystem /usr/include/arm-linux-gnueabihf/qt5/QtSvg  -O2 -g -DNDEBUG   -W -Wall -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -pedantic -Wdouble-promotion -Wformat=2 -Wno-unused-macros -Wno-switch-enum -Wno-unknown-pragmas -Wlogical-op -Wsuggest-override -fPIC -std=c++14 -o CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o -c /home/pi/edb-debugger/src/RegisterViewModelBase.cpp
/home/pi/edb-debugger/src/RegisterViewModelBase.cpp: In member function ‘int RegisterViewModelBase::SIMDFormatItem<StoredType, SizingType>::valueMaxLength() const [with StoredType = edb::detail::value_type80; SizingType = edb::detail::value_type80]’:
/home/pi/edb-debugger/src/RegisterViewModelBase.cpp:934:2: error: static assertion failed: Only 80-bit long doubles are supported
  static_assert(sizeof(edb::value80) <= sizeof(long double), "Only 80-bit long doubles are supported");
  ^~~~~~~~~~~~~
src/CMakeFiles/edb.dir/build.make:806: recipe for target 'src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o' failed
make[2]: *** [src/CMakeFiles/edb.dir/RegisterViewModelBase.cpp.o] Error 1
```
:'(

Sure, every time i clean up the build folder. Aaaand i have another error occurred.
```
pi@raspberrypi:~/edb-debugger $ git checkout
Your branch is up-to-date with 'origin/arm-regression-fixes'.


[ 20%] Building CXX object src/CMakeFiles/edb.dir/capstone-edb/Instruction.cpp.o
cd /home/pi/edb-debugger/build/src && /usr/bin/c++   -DDEFAULT_PLUGIN_PATH=\"/usr/local/lib/edb\" -DEDB_ARM32 -DEDB_IS_32_BIT=true -DEDB_IS_64_BIT=false -DENABLE_GRAPH -DGIT_BRANCH=d7f27571817022c3b5ab8e1fc866ad07e5705692 -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XMLPATTERNS_LIB -DQT_XML_LIB -Dedb_EXPORTS -I/home/pi/edb-debugger/build/src -I/home/pi/edb-debugger/src -I/usr/include/capstone -I/usr/include/graphviz -I/home/pi/edb-debugger/include -I/home/pi/edb-debugger/include/os/unix -I/home/pi/edb-debugger/include/os/unix/linux -I/home/pi/edb-debugger/include/arch/arm-generic -I/home/pi/edb-debugger/src/capstone-edb -I/home/pi/edb-debugger/src/qhexview -I/home/pi/edb-debugger/src/widgets -I/home/pi/edb-debugger/src/session -isystem /usr/include/arm-linux-gnueabihf/qt5 -isystem /usr/include/arm-linux-gnueabihf/qt5/QtWidgets -isystem /usr/include/arm-linux-gnueabihf/qt5/QtGui -isystem /usr/include/arm-linux-gnueabihf/qt5/QtCore -isystem /usr/lib/arm-linux-gnueabihf/qt5/mkspecs/linux-g++ -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXml -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXmlPatterns -isystem /usr/include/arm-linux-gnueabihf/qt5/QtNetwork -isystem /usr/include/arm-linux-gnueabihf/qt5/QtSvg  -O2 -g -DNDEBUG   -W -Wall -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -pedantic -Wdouble-promotion -Wformat=2 -Wno-unused-macros -Wno-switch-enum -Wno-unknown-pragmas -Wlogical-op -Wsuggest-override -fPIC -std=c++14 -o CMakeFiles/edb.dir/capstone-edb/Instruction.cpp.o -c /home/pi/edb-debugger/src/capstone-edb/Instruction.cpp
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::is_simd_register(const CapstoneEDB::Operand&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:46:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type != X86_OP_REG)
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::apriori_not_simd(const CapstoneEDB::Instruction&, const CapstoneEDB::Operand&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:66:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_REG && !is_simd_register(operand))
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:68:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_IMM)
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::KxRegisterPresent(const CapstoneEDB::Instruction&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:79:19: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (op->type == X86_OP_REG && X86_REG_K0 <= op->reg && op->reg <= X86_REG_K7) {
                   ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In member function ‘std::__cxx11::string CapstoneEDB::Formatter::to_string(const CapstoneEDB::Operand&) const’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:464:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_REG) {
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::KxRegisterPresent(const CapstoneEDB::Instruction&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:502:19: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (op->type == X86_OP_REG && X86_REG_K0 <= op->reg && op->reg <= X86_REG_K7)
                   ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::is_SIMD_PD(const CapstoneEDB::Operand&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:852:24: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (insn[2]->type != X86_OP_IMM)
                        ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::modifies_pc(const CapstoneEDB::Instruction&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:1123:9: error: ‘const struct cs_arm_op’ has no member named ‘access’
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
         ^~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:1123:17: error: ‘CS_AC_WRITE’ was not declared in this scope
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
                 ^~~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:1123:41: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum cs_op_type’ [-Wenum-compare]
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
                                         ^~~~~~~~~
src/CMakeFiles/edb.dir/build.make:974: recipe for target 'src/CMakeFiles/edb.dir/capstone-edb/Instruction.cpp.o' failed
make[2]: *** [src/CMakeFiles/edb.dir/capstone-edb/Instruction.cpp.o] Error 1

```

```
cd /home/pi/edb-debugger/build/src && /usr/bin/c++   -DDEFAULT_PLUGIN_PATH=\"/usr/local/lib/edb\" -DEDB_ARM32 -DEDB_IS_32_BIT=true -DEDB_IS_64_BIT=false -DENABLE_GRAPH -DGIT_BRANCH=0f5731186503a00fb40c5c6f351def5cdf830cc1 -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XMLPATTERNS_LIB -DQT_XML_LIB -Dedb_EXPORTS -I/home/pi/edb-debugger/build/src -I/home/pi/edb-debugger/src -I/usr/local/include -I/usr/include/graphviz -I/home/pi/edb-debugger/include -I/home/pi/edb-debugger/include/os/unix -I/home/pi/edb-debugger/include/os/unix/linux -I/home/pi/edb-debugger/include/arch/arm-generic -I/home/pi/edb-debugger/src/capstone-edb -I/home/pi/edb-debugger/src/qhexview -I/home/pi/edb-debugger/src/widgets -I/home/pi/edb-debugger/src/session -isystem /usr/include/arm-linux-gnueabihf/qt5 -isystem /usr/include/arm-linux-gnueabihf/qt5/QtWidgets -isystem /usr/include/arm-linux-gnueabihf/qt5/QtGui -isystem /usr/include/arm-linux-gnueabihf/qt5/QtCore -isystem /usr/lib/arm-linux-gnueabihf/qt5/mkspecs/linux-g++ -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXml -isystem /usr/include/arm-linux-gnueabihf/qt5/QtXmlPatterns -isystem /usr/include/arm-linux-gnueabihf/qt5/QtNetwork -isystem /usr/include/arm-linux-gnueabihf/qt5/QtSvg  -O2 -g -DNDEBUG   -W -Wall -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -pedantic -Wdouble-promotion -Wformat=2 -Wno-unused-macros -Wno-switch-enum -Wno-unknown-pragmas -Wlogical-op -Wsuggest-override -fPIC -std=c++14 -o CMakeFiles/edb.dir/capstone-edb/Instruction.cpp.o -c /home/pi/edb-debugger/src/capstone-edb/Instruction.cpp -H
. /home/pi/edb-debugger/include/Instruction.h
.. /home/pi/edb-debugger/include/../src/capstone-edb/include/Instruction.h
... /home/pi/edb-debugger/include/API.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/QtGlobal
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qglobal.h
...... /usr/include/c++/6/cstddef
....... /usr/include/arm-linux-gnueabihf/c++/6/bits/c++config.h
........ /usr/include/arm-linux-gnueabihf/c++/6/bits/os_defines.h
......... /usr/include/features.h
.......... /usr/include/arm-linux-gnueabihf/sys/cdefs.h
........... /usr/include/arm-linux-gnueabihf/bits/wordsize.h
.......... /usr/include/arm-linux-gnueabihf/gnu/stubs.h
........... /usr/include/arm-linux-gnueabihf/gnu/stubs-hard.h
........ /usr/include/arm-linux-gnueabihf/c++/6/bits/cpu_defines.h
....... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
...... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qconfig.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qfeatures.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qsystemdetection.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qprocessordetection.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qcompilerdetection.h
....... /usr/include/c++/6/utility
........ /usr/include/c++/6/bits/stl_relops.h
........ /usr/include/c++/6/bits/stl_pair.h
......... /usr/include/c++/6/bits/move.h
.......... /usr/include/c++/6/bits/concept_check.h
.......... /usr/include/c++/6/type_traits
........ /usr/include/c++/6/initializer_list
...... /usr/include/c++/6/algorithm
....... /usr/include/c++/6/bits/stl_algobase.h
........ /usr/include/c++/6/bits/functexcept.h
......... /usr/include/c++/6/bits/exception_defines.h
........ /usr/include/c++/6/bits/cpp_type_traits.h
........ /usr/include/c++/6/ext/type_traits.h
........ /usr/include/c++/6/ext/numeric_traits.h
........ /usr/include/c++/6/bits/stl_iterator_base_types.h
........ /usr/include/c++/6/bits/stl_iterator_base_funcs.h
......... /usr/include/c++/6/debug/assertions.h
........ /usr/include/c++/6/bits/stl_iterator.h
......... /usr/include/c++/6/bits/ptr_traits.h
........ /usr/include/c++/6/debug/debug.h
........ /usr/include/c++/6/bits/predefined_ops.h
....... /usr/include/c++/6/bits/stl_algo.h
........ /usr/include/c++/6/cstdlib
......... /usr/include/stdlib.h
.......... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
.......... /usr/include/arm-linux-gnueabihf/bits/waitflags.h
.......... /usr/include/arm-linux-gnueabihf/bits/waitstatus.h
.......... /usr/include/xlocale.h
.......... /usr/include/arm-linux-gnueabihf/sys/types.h
........... /usr/include/arm-linux-gnueabihf/bits/types.h
............ /usr/include/arm-linux-gnueabihf/bits/wordsize.h
............ /usr/include/arm-linux-gnueabihf/bits/typesizes.h
........... /usr/include/time.h
........... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
........... /usr/include/endian.h
............ /usr/include/arm-linux-gnueabihf/bits/endian.h
............ /usr/include/arm-linux-gnueabihf/bits/byteswap.h
............. /usr/include/arm-linux-gnueabihf/bits/byteswap-16.h
........... /usr/include/arm-linux-gnueabihf/sys/select.h
............ /usr/include/arm-linux-gnueabihf/bits/select.h
............ /usr/include/arm-linux-gnueabihf/bits/sigset.h
............ /usr/include/time.h
............ /usr/include/arm-linux-gnueabihf/bits/time.h
........... /usr/include/arm-linux-gnueabihf/sys/sysmacros.h
........... /usr/include/arm-linux-gnueabihf/bits/pthreadtypes.h
.......... /usr/include/alloca.h
........... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
.......... /usr/include/arm-linux-gnueabihf/bits/stdlib-bsearch.h
.......... /usr/include/arm-linux-gnueabihf/bits/stdlib-float.h
........ /usr/include/c++/6/bits/algorithmfwd.h
........ /usr/include/c++/6/bits/stl_heap.h
........ /usr/include/c++/6/bits/stl_tempbuf.h
......... /usr/include/c++/6/bits/stl_construct.h
.......... /usr/include/c++/6/new
........... /usr/include/c++/6/exception
............ /usr/include/c++/6/bits/atomic_lockfree_defines.h
............ /usr/include/c++/6/bits/exception_ptr.h
............ /usr/include/c++/6/bits/nested_exception.h
.......... /usr/include/c++/6/ext/alloc_traits.h
........... /usr/include/c++/6/bits/alloc_traits.h
............ /usr/include/c++/6/bits/memoryfwd.h
........ /usr/include/c++/6/bits/uniform_int_dist.h
......... /usr/include/c++/6/limits
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypeinfo.h
....... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypetraits.h
........ /usr/include/arm-linux-gnueabihf/qt5/QtCore/qglobal.h
....... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qisenum.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qsysinfo.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qlogging.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qflags.h
....... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypeinfo.h
........ /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypetraits.h
........ /usr/include/arm-linux-gnueabihf/qt5/QtCore/qisenum.h
....... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypetraits.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic.h
....... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qbasicatomic.h
........ /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic.h
........ /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic_cxx11.h
......... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qgenericatomic.h
.......... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypeinfo.h
........... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypetraits.h
........... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qisenum.h
......... /usr/include/c++/6/atomic
.......... /usr/include/c++/6/bits/atomic_base.h
........... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stdint.h
............ /usr/include/stdint.h
............. /usr/include/arm-linux-gnueabihf/bits/wchar.h
............. /usr/include/arm-linux-gnueabihf/bits/wordsize.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qglobalstatic.h
....... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qnumeric.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qversiontagging.h
... /home/pi/edb-debugger/include/../src/capstone-edb/include/Formatter.h
.... /usr/include/c++/6/string
..... /usr/include/c++/6/bits/stringfwd.h
..... /usr/include/c++/6/bits/char_traits.h
...... /usr/include/c++/6/bits/postypes.h
....... /usr/include/c++/6/cwchar
........ /usr/include/wchar.h
......... /usr/include/stdio.h
......... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stdarg.h
......... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
...... /usr/include/c++/6/cwchar
....... /usr/include/wchar.h
...... /usr/include/c++/6/cstdint
..... /usr/include/c++/6/bits/allocator.h
...... /usr/include/arm-linux-gnueabihf/c++/6/bits/c++allocator.h
....... /usr/include/c++/6/ext/new_allocator.h
..... /usr/include/c++/6/bits/localefwd.h
...... /usr/include/arm-linux-gnueabihf/c++/6/bits/c++locale.h
....... /usr/include/c++/6/clocale
........ /usr/include/locale.h
......... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
......... /usr/include/arm-linux-gnueabihf/bits/locale.h
...... /usr/include/c++/6/iosfwd
...... /usr/include/c++/6/cctype
....... /usr/include/ctype.h
..... /usr/include/c++/6/bits/ostream_insert.h
...... /usr/include/c++/6/bits/cxxabi_forced.h
..... /usr/include/c++/6/bits/stl_function.h
...... /usr/include/c++/6/backward/binders.h
..... /usr/include/c++/6/bits/range_access.h
..... /usr/include/c++/6/bits/basic_string.h
...... /usr/include/c++/6/ext/atomicity.h
....... /usr/include/arm-linux-gnueabihf/c++/6/bits/gthr.h
........ /usr/include/arm-linux-gnueabihf/c++/6/bits/gthr-default.h
......... /usr/include/pthread.h
.......... /usr/include/sched.h
........... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
........... /usr/include/time.h
........... /usr/include/arm-linux-gnueabihf/bits/sched.h
.......... /usr/include/time.h
........... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
........... /usr/include/arm-linux-gnueabihf/bits/time.h
............ /usr/include/arm-linux-gnueabihf/bits/timex.h
.......... /usr/include/arm-linux-gnueabihf/bits/setjmp.h
.......... /usr/include/arm-linux-gnueabihf/bits/wordsize.h
....... /usr/include/arm-linux-gnueabihf/c++/6/bits/atomic_word.h
...... /usr/include/c++/6/ext/string_conversions.h
....... /usr/include/c++/6/cstdlib
....... /usr/include/c++/6/cwchar
........ /usr/include/wchar.h
....... /usr/include/c++/6/cstdio
........ /usr/include/stdio.h
......... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
......... /usr/include/libio.h
.......... /usr/include/_G_config.h
........... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
........... /usr/include/wchar.h
.......... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stdarg.h
......... /usr/include/arm-linux-gnueabihf/bits/stdio_lim.h
......... /usr/include/arm-linux-gnueabihf/bits/sys_errlist.h
......... /usr/include/arm-linux-gnueabihf/bits/stdio.h
....... /usr/include/c++/6/cerrno
........ /usr/include/errno.h
......... /usr/include/arm-linux-gnueabihf/bits/errno.h
.......... /usr/include/linux/errno.h
........... /usr/include/arm-linux-gnueabihf/asm/errno.h
............ /usr/include/asm-generic/errno.h
............. /usr/include/asm-generic/errno-base.h
...... /usr/include/c++/6/bits/functional_hash.h
....... /usr/include/c++/6/bits/hash_bytes.h
..... /usr/include/c++/6/bits/basic_string.tcc
... /home/pi/edb-debugger/include/../src/capstone-edb/include/Operand.h
.... /usr/local/include/capstone/capstone.h
..... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stdarg.h
..... /usr/include/c++/6/stdlib.h
...... /usr/include/c++/6/cstdlib
..... /usr/local/include/capstone/platform.h
...... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stdbool.h
...... /usr/include/inttypes.h
..... /usr/local/include/capstone/arm.h
..... /usr/local/include/capstone/arm64.h
..... /usr/local/include/capstone/mips.h
..... /usr/local/include/capstone/ppc.h
..... /usr/local/include/capstone/sparc.h
..... /usr/local/include/capstone/systemz.h
..... /usr/local/include/capstone/x86.h
..... /usr/local/include/capstone/xcore.h
... /home/pi/edb-debugger/include/../src/capstone-edb/include/Inspection.h
. /usr/include/arm-linux-gnueabihf/qt5/QtCore/QRegularExpression
.. /usr/include/arm-linux-gnueabihf/qt5/QtCore/qregularexpression.h
... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qstring.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qchar.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qbytearray.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qrefcount.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qnamespace.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qarraydata.h
...... /usr/include/string.h
....... /usr/lib/gcc/arm-linux-gnueabihf/6/include/stddef.h
..... /usr/include/c++/6/stdlib.h
..... /usr/include/c++/6/iterator
...... /usr/include/c++/6/ostream
....... /usr/include/c++/6/ios
........ /usr/include/c++/6/bits/ios_base.h
......... /usr/include/c++/6/bits/locale_classes.h
.......... /usr/include/c++/6/bits/locale_classes.tcc
......... /usr/include/c++/6/system_error
.......... /usr/include/arm-linux-gnueabihf/c++/6/bits/error_constants.h
........... /usr/include/c++/6/cerrno
............ /usr/include/errno.h
.......... /usr/include/c++/6/stdexcept
........ /usr/include/c++/6/streambuf
......... /usr/include/c++/6/bits/streambuf.tcc
........ /usr/include/c++/6/bits/basic_ios.h
......... /usr/include/c++/6/bits/locale_facets.h
.......... /usr/include/c++/6/cwctype
........... /usr/include/wctype.h
............ /usr/include/wchar.h
.......... /usr/include/c++/6/cctype
.......... /usr/include/arm-linux-gnueabihf/c++/6/bits/ctype_base.h
.......... /usr/include/c++/6/bits/streambuf_iterator.h
.......... /usr/include/arm-linux-gnueabihf/c++/6/bits/ctype_inline.h
.......... /usr/include/c++/6/bits/locale_facets.tcc
......... /usr/include/c++/6/bits/basic_ios.tcc
....... /usr/include/c++/6/bits/ostream.tcc
...... /usr/include/c++/6/istream
....... /usr/include/c++/6/bits/istream.tcc
...... /usr/include/c++/6/bits/stream_iterator.h
... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qstringlist.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qlist.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qalgorithms.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qiterator.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qhashfunctions.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qpair.h
...... /usr/include/c++/6/numeric
....... /usr/include/c++/6/bits/stl_numeric.h
..... /usr/include/c++/6/list
...... /usr/include/c++/6/bits/stl_list.h
....... /usr/include/c++/6/bits/allocated_ptr.h
....... /usr/include/c++/6/ext/aligned_buffer.h
...... /usr/include/c++/6/bits/list.tcc
..... /usr/include/c++/6/stdlib.h
..... /usr/lib/gcc/arm-linux-gnueabihf/6/include-fixed/limits.h
...... /usr/lib/gcc/arm-linux-gnueabihf/6/include-fixed/syslimits.h
....... /usr/lib/gcc/arm-linux-gnueabihf/6/include-fixed/limits.h
........ /usr/include/limits.h
......... /usr/include/arm-linux-gnueabihf/bits/posix1_lim.h
.......... /usr/include/arm-linux-gnueabihf/bits/local_lim.h
........... /usr/include/linux/limits.h
......... /usr/include/arm-linux-gnueabihf/bits/posix2_lim.h
......... /usr/include/arm-linux-gnueabihf/bits/xopen_lim.h
.......... /usr/include/arm-linux-gnueabihf/bits/stdio_lim.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qbytearraylist.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qlist.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qstringlist.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qregexp.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qstringmatcher.h
... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qshareddata.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qhash.h
... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qvariant.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qmetatype.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qatomic.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qvarlengtharray.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qcontainerfwd.h
...... /usr/include/c++/6/stdlib.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qisenum.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypetraits.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qobjectdefs.h
...... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qobjectdefs_impl.h
..... /usr/include/c++/6/vector
...... /usr/include/c++/6/bits/stl_uninitialized.h
...... /usr/include/c++/6/bits/stl_vector.h
...... /usr/include/c++/6/bits/stl_bvector.h
...... /usr/include/c++/6/bits/vector.tcc
..... /usr/include/c++/6/map
...... /usr/include/c++/6/bits/stl_tree.h
...... /usr/include/c++/6/bits/stl_map.h
....... /usr/include/c++/6/tuple
........ /usr/include/c++/6/array
........ /usr/include/c++/6/bits/uses_allocator.h
...... /usr/include/c++/6/bits/stl_multimap.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qmap.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qtypetraits.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qstringlist.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qobject.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qscopedpointer.h
...... /usr/include/c++/6/stdlib.h
..... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qobject_impl.h
.... /usr/include/arm-linux-gnueabihf/qt5/QtCore/qbytearraylist.h
. /usr/include/arm-linux-gnueabihf/qt5/QtCore/QRegExp
.. /usr/include/arm-linux-gnueabihf/qt5/QtCore/qregexp.h
. /usr/include/arm-linux-gnueabihf/qt5/QtCore/QString
.. /usr/include/arm-linux-gnueabihf/qt5/QtCore/qstring.h
. /usr/include/arm-linux-gnueabihf/qt5/QtCore/QStringList
.. /usr/include/arm-linux-gnueabihf/qt5/QtCore/qstringlist.h
. /usr/include/c++/6/cassert
.. /usr/include/assert.h
. /usr/include/c++/6/cctype
. /usr/include/c++/6/cstring
. /usr/include/c++/6/sstream
.. /usr/include/c++/6/bits/sstream.tcc
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::is_simd_register(const CapstoneEDB::Operand&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:46:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type != X86_OP_REG)
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::apriori_not_simd(const CapstoneEDB::Instruction&, const CapstoneEDB::Operand&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:66:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_REG && !is_simd_register(operand))
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:68:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_IMM)
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::{anonymous}::KxRegisterPresent(const CapstoneEDB::Instruction&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:79:19: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (op->type == X86_OP_REG && X86_REG_K0 <= op->reg && op->reg <= X86_REG_K7) {
                   ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In member function ‘std::__cxx11::string CapstoneEDB::Formatter::to_string(const CapstoneEDB::Operand&) const’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:464:23: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
  if (operand->type == X86_OP_REG) {
                       ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::KxRegisterPresent(const CapstoneEDB::Instruction&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:502:19: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (op->type == X86_OP_REG && X86_REG_K0 <= op->reg && op->reg <= X86_REG_K7)
                   ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::is_SIMD_PD(const CapstoneEDB::Operand&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:852:24: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum x86_op_type’ [-Wenum-compare]
   if (insn[2]->type != X86_OP_IMM)
                        ^~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp: In function ‘bool CapstoneEDB::modifies_pc(const CapstoneEDB::Instruction&)’:
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:1123:9: error: ‘const struct cs_arm_op’ has no member named ‘access’
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
         ^~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:1123:17: error: ‘CS_AC_WRITE’ was not declared in this scope
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
                 ^~~~~~~~~~~
/home/pi/edb-debugger/src/capstone-edb/Instruction.cpp:1123:41: warning: comparison between ‘const arm_op_type {aka const enum arm_op_type}’ and ‘enum cs_op_type’ [-Wenum-compare]
   if(op.access==CS_AC_WRITE && op.type==CS_OP_REG && op.reg==ARM_REG_PC)
                                         ^~~~~~~~~
Multiple include guards may be useful for:
/home/pi/edb-debugger/include/Instruction.h
/usr/include/arm-linux-gnueabihf/asm/errno.h
/usr/include/arm-linux-gnueabihf/bits/byteswap-16.h
/usr/include/arm-linux-gnueabihf/bits/byteswap.h
/usr/include/arm-linux-gnueabihf/bits/endian.h
/usr/include/arm-linux-gnueabihf/bits/errno.h
/usr/include/arm-linux-gnueabihf/bits/local_lim.h
/usr/include/arm-linux-gnueabihf/bits/locale.h
/usr/include/arm-linux-gnueabihf/bits/sched.h
/usr/include/arm-linux-gnueabihf/bits/select.h
/usr/include/arm-linux-gnueabihf/bits/sigset.h
/usr/include/arm-linux-gnueabihf/bits/stdio.h
/usr/include/arm-linux-gnueabihf/bits/stdlib-bsearch.h
/usr/include/arm-linux-gnueabihf/bits/stdlib-float.h
/usr/include/arm-linux-gnueabihf/bits/sys_errlist.h
/usr/include/arm-linux-gnueabihf/bits/typesizes.h
/usr/include/arm-linux-gnueabihf/bits/waitflags.h
/usr/include/arm-linux-gnueabihf/bits/waitstatus.h
/usr/include/arm-linux-gnueabihf/c++/6/bits/ctype_base.h
/usr/include/arm-linux-gnueabihf/c++/6/bits/ctype_inline.h
/usr/include/arm-linux-gnueabihf/gnu/stubs-hard.h
/usr/include/arm-linux-gnueabihf/gnu/stubs.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/QRegExp
/usr/include/arm-linux-gnueabihf/qt5/QtCore/QRegularExpression
/usr/include/arm-linux-gnueabihf/qt5/QtCore/QString
/usr/include/arm-linux-gnueabihf/qt5/QtCore/QStringList
/usr/include/arm-linux-gnueabihf/qt5/QtCore/QtGlobal
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qbasicatomic.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qcompilerdetection.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qconfig.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qfeatures.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qflags.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qglobalstatic.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qlogging.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qprocessordetection.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qstringlist.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qsysinfo.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qsystemdetection.h
/usr/include/arm-linux-gnueabihf/qt5/QtCore/qversiontagging.h
/usr/include/assert.h
/usr/include/c++/6/cassert
/usr/include/c++/6/clocale
/usr/include/c++/6/cstdio
/usr/include/c++/6/cstring
/usr/include/c++/6/cwctype
/usr/include/limits.h
/usr/include/linux/errno.h
/usr/lib/gcc/arm-linux-gnueabihf/6/include-fixed/syslimits.h
```

--- a/plugins/DebuggerCore/unix/linux/arch/arm-generic/PlatformState.cpp
+++ b/plugins/DebuggerCore/unix/linux/arch/arm-generic/PlatformState.cpp
@@ -292,8 +292,9 @@ void PlatformState::set_register(const QString &name, edb::reg_t value) {
  */
 Register PlatformState::gp_register(size_t n) const {
 #ifdef EDB_ARM32
-	assert(n < GPR::GPRegNames.size());
-	return make_Register<32>(gpr.GPRegNames[n].front(), gpr.GPRegs[n], Register::TYPE_GPR);
+	if(n < GPR::GPRegNames.size())
+		return make_Register<32>(gpr.GPRegNames[n].front(), gpr.GPRegs[n], Register::TYPE_GPR);
+	return Register();
 #else
 	return Register(); // FIXME: stub
 #endif

